File: hackerrank.com/problem-solving/apple_and_orange.js
// https://www.hackerrank.com/challenges/apple-and-orange/problem

function countApplesAndOranges(s, t, a, b, apples, oranges) {
  const applePoint = apples.map(app => a + app).filter(app => app >= s && app <= t)
  const orang = oranges.map(org => b + org).filter(org => org >= s && org <= t)

  console.log(applePoint.length)
  console.log(orang.length)
}


function main(inputData) {
  const lines = inputData.split("\n");
  const [s, t] = lines[0].split(' ').map(x => parseInt(x));
  const [a, b] = lines[1].split(' ').map(x => parseInt(x));
  const [m, n] = lines[2].split(' ').map(x => parseInt(x));
  const apples = lines[3].split(' ').map(x => parseInt(x));
  const oranges = lines[4].split(' ').map(x => parseInt(x));
  
  countApplesAndOranges(s, t, a, b, apples, oranges);
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
let _input = '';
process.stdin.on("data", input => {
  _input += input;
});

process.stdin.on("end", () => {
 main(_input);
});

File: hackerrank.com/problem-solving/apple_and_orange.py
# https://www.hackerrank.com/challenges/apple-and-orange/problem

def countApplesAndOranges(s, t, a, b, apples, oranges):
    print(sum([1 for app in apples if t >= a + app >= s]))
    print(sum([1 for org in oranges if t >= b + org >= s]))

def countApplesAndOranges2(s, t, a, b, apples, oranges):
    print(sum(t >= a + app >= s for app in apples))
    print(sum(t >= b + app >= s for app in oranges))

if __name__ == '__main__':
    s, t = map(int, input().split())
    a, b = map(int, input().split())
    n, m = map(int, input().split())
    apples = list(map(int, input().split()))
    oranges = list(map(int, input().split()))

    countApplesAndOranges(s, t, a, b, apples, oranges)
    

